chips/sam4l/src/eic.rs

```diff
@@ -0,0 +1,321 @@
+// ICR: Writing a one to this bit will clear the corresponding bit in ISR.
+// MODE:    0: The external interrupt is edge triggered.
+//          1: The external interrupt is level triggered.'
+// EDGE:    0: The external interrupt triggers on falling edge.
+//          1: The external interrupt triggers on rising edge.
+// LEVEL:   0: The external interrupt triggers on low level.
+//          1: The external interrupt triggers on high level.
+// FILTER:  0: The external interrupt is not filtered.
+//          1: The external interrupt is filtered.
+// ASYNC:   0: The external interrupt is synchronized to CLK_SYNC.
+//          1: The external interrupt is asynchronous.
+// EN: Writing a one to this bit will enable the corresponding external interrupt.
+// DIS: Writing a one to this bit will disable the corresponding external interrupt.
+// CTRL:    0: The corresponding external interrupt is disabled.
+//          1: The corresponding external interrupt is enabled.
+
+register_bitfields![
+    u32,
+    Interrupt [
+        INT OFFSET(0) NUMBITS(32) []
+    ],
+    // Test is not being used right now
+    Test [
+        //0: This bit disables external interrupt test mode.
+        //1: This bit enables external interrupt test mode.
+        TESTEN OFFSET(31) NUMBITS(1) [],
+
+        // Writing a zero to this bit will set the input value to INTn to zero, if test mode is enabled. 
+        // Writing a one to this bit will set the input value to INTn to one, if test mode is enabled.
+        INT OFFSET(0) NUMBITS(31) []
+    ]
+];
+
+// Page 59 of SAM4L data sheet
+const EIC_BASE: StaticRef<EicRegisters> =
+    unsafe { StaticRef::new(0x400F1000 as *const EicRegisters) };
+
+pub struct Eic<'a> {
+    registers: StaticRef<EicRegisters>,
+    enabled: Cell<bool>,
+    callbacks: [OptionalCell<&'a dyn hil::eic::Client>; 9],
+}
+
+impl<'a> hil::eic::ExternalInterruptController for Eic<'a> {
+    type Line = Line;
+
+    fn line_enable(
+        &self,
+        line: &Self::Line,
+        int_mode: hil::eic::InterruptMode,
+        filter_mode: hil::eic::FilterMode,
+        syn_mode: hil::eic::SynchronizationMode,
+    ) {
+        if !self.is_enabled() {
+            self.enable();
+        }
+
+        let interrupt_line = *line as u32;
+        let regs: &EicRegisters = &*self.registers;
+
+        regs.en.write(Interrupt::INT.val(interrupt_line));
+
+        self.line_configure(interrupt_line, int_mode, filter_mode, syn_mode);
+        self.line_enable_interrupt(interrupt_line);
+    }
+
+    fn line_disable(&self, line: &Self::Line) {
+        if !self.is_enabled() {
+            return;
+        }
+
+        let interrupt_line: u32 = 1 << *line as u32;
+        let regs: &EicRegisters = &*self.registers;
+        regs.dis.write(Interrupt::INT.val(interrupt_line));
+        self.line_disable_interrupt(interrupt_line);
+    }
+}
+
+impl<'a> Eic<'a> {
+    pub fn line_configure(
+        &self,
+        interrupt_line: u32,
```

Comment:
I think you can change this to
```suggestion
        line: Line,
```